src/main/java/server/ChatServer.java

```
@@ -142,82 +99,113 @@ public static void onlineMessage() throws IOException {
     Socket s;
     boolean isloggedin;
 
+
+    /**
+     * Try to accept the connection. Returns false if the connection is rejected. for any reason.
+     * @param userService
+     * @return
+     */
+    static ClientHandler tryAccept(Socket s, UserService userService) {
+        DataInputStream dis = new DataInputStream(s.getInputStream());
+        PrintWriter dos = new PrintWriter(s.getOutputStream(),true);
+        StringTokenizer st = new StringTokenizer(dis.readLine(), "#");
+        String cmd = st.nextToken();
+
+        try {
+            if (cmd.equals("CONNECT")) {
+                String username = st.nextToken();
+                if (!userService.usernameExists(username)) {
+                    dos.println("CLOSE#2");
+                    s.close();
+                    break;
+                }
+
+                return new ClientHandler(s, usenrame, dis, dos);
+            }
+
+            dos.println("CLOSE#1");
+        } catch (Exception e) {
```

I don't recommend swallowing exceptions and doing nothing, but it's good enough